sources/HUBCollectionView.m

```diff
@@ -42,6 +46,27 @@ - (void)setContentOffset:(CGPoint)contentOffset
     [super setContentOffset:contentOffset];
 }
 
+- (NSMutableSet<NSString *> *)registeredCollectionViewCellReuseIdentifiers
+{
+    if (!_registeredCollectionViewCellReuseIdentifiers) {
+        _registeredCollectionViewCellReuseIdentifiers = [NSMutableSet new];
+    }
+
+    return _registeredCollectionViewCellReuseIdentifiers;
+}
+
+- (__kindof UICollectionViewCell *)dequeueReusableCellWithReuseIdentifier:(NSString *)identifier
+                                                             forIndexPath:(NSIndexPath *)indexPath
+                                                cellClassWhenUnregistered:(Class)cellClass
+{
+    if (![self.registeredCollectionViewCellReuseIdentifiers containsObject:identifier]) {
+        [self registerClass:cellClass forCellWithReuseIdentifier:identifier];
```

Comment:
Not sure I follow here, we don't know all the potential identifiers in `loadCollectionViewIfNeeded`. This code is just copied over from `collectionView:cellForItemAtIndexPath:` which is where it's being called from still.